Sources/Snapshot.swift

```diff
@@ -0,0 +1,38 @@
+/// Represents a snapshot of a collection.
+///
+/// The changeset associated with the first snapshot received by any given observation
+/// is generally irrelevant. Observations should special case the first snapshot as a
+/// complete replacement, or whatever semantic that fits their purpose.
+///
+/// The previous version of the collection is unavailable on the initial snapshot,
+/// since there is no history to refer to.
+public struct Snapshot<Collection: Swift.Collection>: SnapshotProtocol {
+	/// The previous version of the collection, or `nil` if `self` is an initial snapshot.
+	public let previous: Collection?
+
+	/// The current version of the collection.
+	public let current: Collection
+
+	/// The changeset which, when applied on `previous`, reproduces `current`.
+	public let changeset: Changeset
```

Comment:
We would end up having equivalently two possibilities even if we pack it as an enum: presence of `previous`, or no `previous`.

In the end, if one observes a _hot_ collection snapshot signal for whatever reason, they _cannot_ rely on the presence of `previous` to determine if the snapshot is an initial one. This further implies that generic algorithms that work for both hot and cold signals cannot rely on this either.

So packing it as an enum seems not having a particular advantage.